crates/ide_assists/src/handlers/reorder_impl.rs

```diff
@@ -75,13 +74,18 @@ pub(crate) fn reorder_impl(acc: &mut Assists, ctx: &AssistContext) -> Option<()>
     }
 
     let target = items.syntax().text_range();
-    acc.add(AssistId("reorder_impl", AssistKind::RefactorRewrite), "Sort methods", target, |edit| {
-        let mut rewriter = algo::SyntaxRewriter::default();
-        for (old, new) in methods.iter().zip(&sorted) {
-            rewriter.replace(old.syntax(), new.syntax());
-        }
-        edit.rewrite(rewriter);
-    })
+    acc.add(
+        AssistId("reorder_impl", AssistKind::RefactorRewrite),
+        "Sort methods",
+        target,
+        |builder| {
+            for (old, new) in
+                methods.into_iter().zip(sorted).filter(|(field, sorted)| field != sorted)
```

Comment:
Why do we need this != condition?

Reply:
We actually don't now that you mention it. I thought this would reduce the produced diff but we are diffing the trees anyways when commiting the work so this is unneeded(did the same in reorder_fields I think which is why I did it here). Will remove that filter